test/test_core.jl

```diff
@@ -187,11 +187,26 @@ end
         @test v_init[1] == 2
         @test v === v_init
 
+        # Julia 0.7 with --depwarn=error:
+        is_deperror07 = VERSION >= v"0.7-" && Base.JLOptions().depwarn == 2
+
         v = randn(3)
-        @set! v[:] = 1
+        @test_skip @set! v[:] .= 0
```

Comment:
`@test_broken` requires the expression to return `false` but `@set! v[:] .= 0` actually throws at the moment. `@test_skip` doesn't execute the body so effectively it's commenting out the expression. I thought it's slightly better than commenting out since it shows up in the test result and we can remember that there is something to fix. But `@test_skip` is not super correct to use it here since `@set! v[:] .= 0` will not return `true`.  I suppose I should have leave informative comments there.